src/popup/components/Settings/RateOfPay.tsx

```diff
@@ -121,31 +117,38 @@ const RateOfPayInput = ({
   maxRateOfPay,
   disabled,
 }: RateOfPayInputProps) => {
-  const rate = React.useMemo(() => {
-    const r = Number(rateOfPay) / 10 ** walletAddress.assetScale;
-    const roundedR = roundWithPrecision(r, walletAddress.assetScale);
+  const t = useTranslation();
+  const [errorMessage, setErrorMessage] = React.useState('');
+
+  const formatAmount = React.useMemo(
+    () => (value: number | string) => {
+      const r = Number(value) / 10 ** walletAddress.assetScale;
+      const roundedR = roundWithPrecision(r, walletAddress.assetScale);
 
-    return formatNumber(roundedR, walletAddress.assetScale, true);
-  }, [rateOfPay, walletAddress.assetScale]);
+      return formatNumber(roundedR, walletAddress.assetScale, true);
+    },
+    [walletAddress.assetScale],
+  );
 
   return (
-    <div className="space-y-2">
-      <Label className="px-2 text-base font-medium text-medium">
-        Rate of pay per hour
-      </Label>
-      <Slider
-        onChange={onRateChange}
-        min={Number(minRateOfPay)}
-        max={Number(maxRateOfPay)}
-        step={Number(minRateOfPay)}
-        value={Number(rateOfPay)}
-        disabled={disabled}
+    <>
+      <InputAmount
+        id="rateOfPay"
+        className="max-w-56"
+        label="Rate of pay per hour"
+        walletAddress={walletAddress}
+        onChange={(rate) => {
+          setErrorMessage('');
+          onRateChange(rate);
+        }}
+        onError={(error) => setErrorMessage(t(error))}
+        errorMessage={errorMessage}
+        min={Number(formatAmount(minRateOfPay))}
+        max={Number(formatAmount(maxRateOfPay))}
+        amount={rateOfPay}
+        controls={true}
```

Comment:
no `controls` for `Budget` input amount as well?

Reply:
I would say only for the rate of pay. The controls do not really fit normal input amounts (`amount` field in the setup screen or in the `budget` tab)

Reply:
Agree with Radu! Usually the user has a clear idea about the amount , an input field makes sense. The rate of pay on the other hand is small. 
@sidvishnoi This looks good. No changes required for the amount.